src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs

```diff
@@ -2860,6 +2860,88 @@ public static void DCS_BasicRoundTripResolveEnumStructTypes()
         SerializationTestTypes.ComparisonHelper.CompareRecursively(value, actual);
     }
 
+    [Fact]
+    public static void DCS_BasicRoundtripDCRVariation1()
+    {
+        SerializationTestTypes.DCRVariations dcrVariationsGoing = new SerializationTestTypes.DCRVariations();
+        dcrVariationsGoing.unknownType1 = new SerializationTestTypes.Person();
+        dcrVariationsGoing.unknownType2 = new SerializationTestTypes.SimpleDC();
+        var setting1 = new DataContractSerializerSettings() { DataContractResolver = new SerializationTestTypes.SimpleResolver_Ser(), MaxItemsInObjectGraph = int.MaxValue, IgnoreExtensionDataObject = false, PreserveObjectReferences = true};
+        var setting2 = new DataContractSerializerSettings() { DataContractResolver = new SerializationTestTypes.SimpleResolver_DeSer(), MaxItemsInObjectGraph = int.MaxValue, IgnoreExtensionDataObject = false, PreserveObjectReferences = true};
+        var dcs1 = new DataContractSerializer(typeof(SerializationTestTypes.CustomClass), setting1);
+        var dcs2 = new DataContractSerializer(typeof(SerializationTestTypes.CustomClass), setting2);
+
+        MemoryStream ms = new MemoryStream();
+        dcs1.WriteObject(ms, dcrVariationsGoing);
+        ms.Position = 0;
```

Comment:
As the method 'CompareRecursively' will verify the object type, properties, fields of the original object & deserialized object. I think we don't need to verify the output from ms. If you think we have the necessary, I will add the baseline to testing.